vm/message/src/lib.rs

```diff
@@ -46,4 +46,6 @@ pub trait Message {
     fn set_gas_fee_cap(&mut self, cap: TokenAmount);
     /// sets the gas premium
     fn set_gas_premium(&mut self, prem: TokenAmount);
+    /// semantic validation of the message
+    fn valid_for_block_inclusion(&self, min_gas: i64) -> Result<(), String>;
```

Comment:
Is there a reason this needs to be part of the message trait? Isn't this only needed for unsigned messages?

Reply:
It will be used on signed messages in the Mpool

Reply:
Is there a reason against putting it in the trait?

Reply:
alrighty, but the check only needs the unsigned message, I'm fine with how it is, but consider just calling `.message()` on the signed message when you need this instead of expanding this trait (already getting bulky and this doesn't seem like it should be part of the trait).

Reply:
> Is there a reason against putting it in the trait?

not syntactically, but logically it doesn't make sense to me, might be opinionated

Reply:
hehe, i like it in the trait more. im gunna leave it for now.